arch/xtensa/core/mpu.c

```diff
@@ -443,6 +443,7 @@ static int mpu_map_region_add(struct xtensa_mpu_map *map,
 
 				xtensa_mpu_entry_set(entry_slot_s, start_addr, true,
 						     access_rights, memory_type);
+				*first_idx = XTENSA_MPU_NUM_ENTRIES - 1;
```

Comment:
```
if (first_idx != NULL) {
```

Reply:
Fixed this by moving assignment to `end` label, with check there.